packages/ref/src/cmip_ref/datasets/obs4mips.py

```diff
@@ -105,9 +109,11 @@ def parse_obs4mips(file: str) -> dict[str, Any | None]:
         )
         return info
 
-    except TypeError:
+    except (TypeError, AttributeError):
+        logger.warning(traceback_message)
```

Comment:
Because you are re-using Python built-in exceptions (`TypeError` and `AttributeError`), they can also be raised by other things than you explicitly raising them, and then the `traceback_message` variable would not be set, which will raise another `NameError`, which will be a bit confusing. This problem also existed previously with only the `TypeError`, so obviously not super serious, but maybe worth to fix.

One way to deal with this is to instead use the `args` attribute that `TypeError` and `AttributeError` always have:
```suggestion
    except (TypeError, AttributeError) as err:
        logger.warning(str(err.args))
```
Note that it will look a bit different because a `traceback_message = "example"` will lead to `str(err.args) = '("example",)'`. If you want to avoid that, you have to do some dance with `if(len(err.args)) == 1: str(err.args[0]) else str(err.args)` or so because if this is an error from something else you might not have args at all, or more than one arg.

Reply:
resolved in my latest commit f02537c